### PR TITLE
chore(as): fix ci lint and set attributes with multierror

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

do the following efforts on AS service:
- fix CI lint in all files
- use **availability_zones** instead of **available_zones**
- set attributes with multierror package in as_group and as_policy

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run=TestAccASGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run=TestAccASGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccASGroup_basic
=== PAUSE TestAccASGroup_basic
=== CONT  TestAccASGroup_basic
--- PASS: TestAccASGroup_basic (181.18s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        181.260s

$ make testacc TEST='./huaweicloud/services/acceptance/as' TESTARGS='-run=TestAccASPolicy_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run=TestAccASPolicy_basic -timeout 360m -parallel 4
=== RUN   TestAccASPolicy_basic
=== PAUSE TestAccASPolicy_basic
=== CONT  TestAccASPolicy_basic
--- PASS: TestAccASPolicy_basic (44.67s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        44.722s
```

### CODE REVIEW

File: huaweicloud/services/as/resource_huaweicloud_as_configuration.go [hunk 1]
```diff
@@ -215,24 +215,32 @@ func ResourceASConfiguration() *schema.Resource {
 	}
 }
 
-func getDisk(diskMeta []interface{}) ([]configurations.DiskOpts, error) {
+func validateDiskSize(diskSize int, diskType string) error {
+	if diskType == "SYS" {
+		if diskSize < 40 || diskSize > 32768 {
+			return fmt.Errorf("For system disk size should be [40, 32768]")
```
Comment (Lance52259) [inline huaweicloud/services/as/resource_huaweicloud_as_configuration.go]: lowercase for the first character of the error message.
```suggestion
			return fmt.Errorf("for system disk size should be [40, 32768]")
```
maybe [40, 32768] is too mathematical.
```suggestion
			return fmt.Errorf("the system disk size should be between 40 and 32768")
```

File: huaweicloud/services/as/resource_huaweicloud_as_configuration.go [hunk 2]
```diff
@@ -340,30 +341,31 @@ func resourceASConfigurationCreate(ctx context.Context, d *schema.ResourceData,
 		InstanceConfig: instanceConfig,
 	}
 
-	logp.Printf("[DEBUG] Create AS configuration Options: %#v", createOpts)
+	log.Printf("[DEBUG] Create AS configuration Options: %#v", createOpts)
 	asConfigId, err := configurations.Create(asClient, createOpts).Extract()
 	if err != nil {
-		return diag.Errorf("Error creating ASConfiguration: %s", err)
+		return diag.Errorf("Error creating AS configuration: %s", err)
```
Comment (Lance52259) [inline huaweicloud/services/as/resource_huaweicloud_as_configuration.go]: ditto. And there are lots of problems like this for error returns.

File: huaweicloud/services/as/resource_huaweicloud_as_group.go [hunk 3]
```diff
@@ -218,58 +220,55 @@ func ResourceASGroup() *schema.Resource {
...
 
-type Network struct {
-	ID string
-}
-
-type Group struct {
-	ID string
-}
+func buildNetworksOpts(d *schema.ResourceData) []groups.NetworkOpts {
+	networks := d.Get("networks").([]interface{})
```
Comment (Lance52259) [inline huaweicloud/services/as/resource_huaweicloud_as_group.go]: how about this
```suggestion
func buildNetworksOpts(networks []interface{}) []groups.NetworkOpts {
```

File: huaweicloud/services/as/resource_huaweicloud_as_group.go [hunk 4]
```diff
@@ -337,11 +304,25 @@ func getAllLBaaSListeners(d *schema.ResourceData, meta interface{}) []groups.LBa
 	return aslisteners
 }
 
+func expandGroupsTags(tagmap map[string]interface{}) []tags.ResourceTag {
+	var taglist []tags.ResourceTag
```
Comment (Lance52259) [inline huaweicloud/services/as/resource_huaweicloud_as_group.go]:
```suggestion
	taglist := make([]tags.ResourceTag, 0, len(tagMap))
```

File: huaweicloud/services/as/resource_huaweicloud_as_group.go [hunk 5]
```diff
@@ -741,9 +709,8 @@ func resourceASGroupDelete(ctx context.Context, d *schema.ResourceData, meta int
 		return diag.Errorf("error creating autoscaling client: %s", err)
 	}
 
-	timeout := d.Timeout(schema.TimeoutDelete)
 	groupID := d.Id()
-	logp.Printf("[DEBUG] Begin to delete AS group %s", groupID)
+	timeout := d.Timeout(schema.TimeoutDelete)
```
Comment (Lance52259) [inline huaweicloud/services/as/resource_huaweicloud_as_group.go:699]: timout is used only once. we don't need to define a variable.

File: huaweicloud/services/as/resource_huaweicloud_as_configuration.go [hunk 6]
```diff
@@ -374,20 +376,21 @@ func resourceASConfigurationDelete(_ context.Context, d *schema.ResourceData, me
 		return diag.Errorf("Error creating autoscaling client: %s", err)
 	}
 
-	groups, err := getASGroupsByConfiguration(asClient, d.Id())
+	configId := d.Id()
+	groups, err := getASGroupsByConfiguration(asClient, configId)
 	if err != nil {
-		return diag.Errorf("Error getting AS groups by configuration ID %q: %s", d.Id(), err)
+		return diag.Errorf("Error getting AS groups by configuration ID %s: %s", configId, err)
```
Comment (hmbig2) [inline huaweicloud/services/as/resource_huaweicloud_as_configuration.go]:
```suggestion
		return diag.Errorf("error getting AS groups by configuration ID %s: %s", configId, err)
```

File: huaweicloud/services/as/resource_huaweicloud_as_policy.go [hunk 7]
```diff
@@ -135,44 +136,40 @@ func getCurrentUTCwithoutSec() string {
...
 	alarmId := d.Get("alarm_id").(string)
-	logp.Printf("[DEBUG] validateParameters alarmId is :%s", alarmId)
-	logp.Printf("[DEBUG] validateParameters policyType is :%s", policyType)
 	scheduledPolicy := d.Get("scheduled_policy").([]interface{})
-	logp.Printf("[DEBUG] validateParameters scheduledPolicy is :%#v", scheduledPolicy)
+
 	if policyType == "ALARM" {
 		if alarmId == "" {
-			return fmtp.Errorf("Parameter alarm_id should be set if policy type is ALARM.")
+			return fmt.Errorf("Parameter alarm_id should be set if policy type is ALARM.")
```
Comment (hmbig2) [inline huaweicloud/services/as/resource_huaweicloud_as_policy.go]:
```suggestion
			return fmt.Errorf("parameter alarm_id should be set if policy type is ALARM")
```

File: huaweicloud/services/as/resource_huaweicloud_as_configuration.go [hunk 8]
```diff
@@ -287,51 +295,44 @@ func getPublicIps(publicIpMeta map[string]interface{}) configurations.PublicIpOp
...
 }
 
-func getInstanceConfig(configDataMap map[string]interface{}) (configurations.InstanceConfigOpts, error) {
+func buildInstanceConfig(configDataMap map[string]interface{}) (configurations.InstanceConfigOpts, error) {
 	disksData := configDataMap["disk"].([]interface{})
-	disks, err := getDisk(disksData)
+	disks, err := buildDiskOpts(disksData)
 	if err != nil {
-		return configurations.InstanceConfigOpts{}, fmtp.Errorf("Error happened when validating disk size: %s", err)
+		return configurations.InstanceConfigOpts{}, fmt.Errorf("the disk size is invalid: %s", err)
```
Comment (hmbig2) [inline huaweicloud/services/as/resource_huaweicloud_as_configuration.go:302]:
```suggestion
		return nil, fmt.Errorf("the disk size is invalid: %s", err)
```

File: huaweicloud/services/as/resource_huaweicloud_as_configuration.go (hunk 1 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_configuration.go]: fixed

File: huaweicloud/services/as/resource_huaweicloud_as_configuration.go (hunk 2 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_configuration.go]: fixed

File: huaweicloud/services/as/resource_huaweicloud_as_group.go (hunk 3 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_group.go]: fixed

File: huaweicloud/services/as/resource_huaweicloud_as_group.go (hunk 4 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_group.go]: fixed

File: huaweicloud/services/as/resource_huaweicloud_as_group.go (hunk 5 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_group.go:699]: I think we can keep it as is

File: huaweicloud/services/as/resource_huaweicloud_as_configuration.go (hunk 6 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_configuration.go]: fixed

File: huaweicloud/services/as/resource_huaweicloud_as_policy.go (hunk 7 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_policy.go]: fixed

File: huaweicloud/services/as/resource_huaweicloud_as_configuration.go (hunk 8 above)
Comment (ShiChangkuo) [reply huaweicloud/services/as/resource_huaweicloud_as_configuration.go:302]: the return value is not a pointer, I think we can keep it as is.

**Pull request merged**